plugins/optimization-detective/class-od-preload-link-collection.php

```diff
@@ -166,6 +166,37 @@ public function get_html(): string {
 		return implode( '', $link_tags );
 	}
 
+	/**
+	 * Gets the HTTP Link header string.
+	 *
+	 * @return string HTTP Link header.
+	 */
+	public function get_headers(): string {
+		$link_headers = array();
+
+		foreach ( $this->get_adjacent_deduplicated_links() as $link ) {
+			$media_features = array( 'screen' );
+			if ( null !== $link['minimum_viewport_width'] && $link['minimum_viewport_width'] > 0 ) {
+				$media_features[] = sprintf( '(min-width: %dpx)', $link['minimum_viewport_width'] );
+			}
+			if ( null !== $link['maximum_viewport_width'] && PHP_INT_MAX !== $link['maximum_viewport_width'] ) {
+				$media_features[] = sprintf( '(max-width: %dpx)', $link['maximum_viewport_width'] );
+			}
+			$link['attributes']['media'] = implode( ' and ', $media_features );
+
+			$link_header = '<' . esc_url( $link['attributes']['href'] ?? '' ) . '>; rel="preload"';
+			foreach ( $link['attributes'] as $name => $value ) {
+				if ( 'href' !== $name ) {
+					$link_header .= sprintf( '; %s="%s"', $name, esc_attr( $value ) );
```

Comment:
Use of `esc_attr()` isn't right here since this is not an HTML context.  I believe you should instead percent-encode the value here using `rawurlencode()`.